src/Streamly/Internal/LZ4/Config.hs

```diff
@@ -104,6 +125,32 @@ removeUncompressedSize size config =
         , compSizeOffset = compSizeOffset config - 4
         }
 
+-- | Stop when end mark is reached.
+addEndMark :: Int32 -> Config s -> Config ('EndMark ': s)
```

Comment:
We can call it addFooter or addTrailer as it would include both endmark and checksum handling.

Reply:
I've simplified it a lot. We can keep on calling it `addEndMark`